Feature/movewait

Shayan-Bathaee opened the pull request:
Main changes:

- Addition of `package.py` and `subpackage.py` in `openTap/UR_Prototype`
- Modifications to message receiving component of `send_request_movement()` in `UR3e.py`
- Modifications to message receiving component of `send_request_from_file()` in `UR3e.py`

Shawn-Armstrong commented:
## Modifications
1. Remove licensing comments from both `package.py` and `subpackage.py`. The comments were initially included for liability purposes when shared with the Universal Robots' community, but they are no longer relevant.

## Observed Bugs
1. **Skipped movement at low speeds**
    - Steps to reproduce:
        1. Program the robot to transition between three positions twice.
        2. Set the speed of position 2 to 0.1 in the second cycle.
    - Expected behavior: The robot should reach all three positions in both cycles.
    - Observed behavior: In the second cycle, the robot skips position 2.
         
       <kbd>![bug1](https://user-images.githubusercontent.com/80125540/234164330-653cd609-3c5c-4754-82d6-4bd5b0d4182a.gif)</kbd>
    - Suggested action: Resolve corner case or set verdict to fail and provide log messages indicating reason. 
       
       I recommend the former, variable `q_actual` chases `q_target` and they'll converge at the same value only at the end of the movement.
      
      ```Python
        # send command
        
        is_finished = False
        while is_finished == False:
          
          # Get package
          
          q_actual_list = []
          q_target_list = []
            for i in range(1, 7):
                q_actual = getattr(joint_data.subpackage_variables, f"Joint{i}_q_actual")
                q_actual_list.append(round(q_actual, 5))
                q_target = getattr(joint_data.subpackage_variables, f"Joint{i}_q_target")
                q_target_list.append(round(q_target, 5))
                
          if q_actual_list == q_target_list:
            is_finished = True
      ```

Shayan-Bathaee commented:
Revisions made: 

- Changed base branch from `main` to `development`
- Removed header comments in `package.py` and `subpackage.py`
- Removed timeout feature in `UR3e.py` within the functions `send_request_movement()` and `send_request_from_file()`
- Added breakout feature in while loop of `send_request_movement()` and `send_request_from_file()` that detects if nothing was received from the socket connection and yields an error if so.

Shawn-Armstrong commented:
I tested it extensively, the aforementioned bug was resolved and all commands reach their waypoint as anticipated. 

It's ready to be merged into development.
